### PR TITLE
cli: Introduct `tilt ci` for waiting until pods are healthy

### PR DESCRIPTION
**Pull request by nicks**
Hello @jazzdan, @maiamcc,

Please review the following commits I made in branch nicks/ch5914/ci:

4b9e391747100f15ed52308ebe0627d19f2d018f (2020-04-01 11:38:30 -0400)
cli: Introduct `tilt ci` for waiting until pods are healthy

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/cli/ci.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+func (c *ciCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "ci [<tilt flags>] [-- <Tiltfile args>]",
+		DisableFlagsInUseLine: true,
+		Short:                 "Start Tilt in CI/batch mode with the given Tiltfile args",
+		Long: `
+Starts Tilt and runs services defined in the Tiltfile.
+
+Exits with failure if any build fails or any server crashes.
```
Comment (maiamcc) [inline internal/cli/ci.go:33]: seems like the main difference between this and `--watch=false` (or whatever it's called now) is that this exits after pods are up and ready, and the latter exists after the final yaml apply is done. maybe call that out in the help text? (probably don't bother contrasting it to `--watch=false` but DO say when this command is "done".)

File: internal/cli/ci.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+`,
+	}
+
+	addWebServerFlags(cmd)
+
+	cmd.Flags().StringVar(&updateModeFlag, "update-mode", string(buildcontrol.UpdateModeAuto),
```
Comment (maiamcc) [inline internal/cli/ci.go:39]: do we expect this mode to run any updates after the initial deploy?

File: internal/cli/ci.go (hunk 1 above)
Comment (nicks) [reply internal/cli/ci.go:33]: For this particular documentation, I think we want to assume the user has never heard of --watch=false. Given that assumption, what do you hope to accomplish by introducing it here?

If we learn that `tilt ci` is a better fit for what most people need, then we don't want to encourage them to learn about --watch=false

File: internal/cli/ci.go (hunk 2 above)
Comment (nicks) [reply internal/cli/ci.go]: good point, removed

File: internal/cli/ci.go (hunk 1 above)
Comment (nicks) [reply internal/cli/ci.go:32]: oh, nm, i misread what you were saying, added some more text on exit conditions

**Comment by victorwuky** [comment]
@nicks : Is this net new in v0.13.0? Seems like something we should blog about. I can start that.

**Comment by nicks** [comment]
@victorwuky yup! I wanted to get it into a few trusted customer hands. They filed some bugs, which we fixed and are going to be in the next release. i can file a story about the blogpost

**Comment by victorwuky** [comment]
Thanks @nicks. I created https://app.clubhouse.io/windmill/story/6051/blog-post-on-tilt-ci

**Pull request merged**